catalyst/contrib/dl/callbacks/tiled_inference.py

```diff
@@ -0,0 +1,242 @@
+from collections import namedtuple
+from pathlib import Path
+
+import numpy as np
+
+import torch
+import torch.nn.functional as F  # noqa: N812, WPS301
+
+from catalyst.contrib.data.cv.datasets.image import Edges, Shape
+from catalyst.core import Callback, CallbackOrder, State
+
+Allocated = namedtuple("Allocated", ["storage", "norm_mask", "weights"])
+
+
+def pyramid_weights(height: int, width: int) -> torch.Tensor:
+    """
+    Computes a weight matrix that assigns bigger weights
+    on pixels at center and less weight to pixels on image boundary.
+    This weight matrix then used for merging individual tile predictions
+    and helps dealing with prediction artifacts on tile boundaries.
+
+    Args:
+        height: Tile height
+        width: Tile width
+
+    Returns:
+        Weight matrix of shape [`height` x `width`]
+    """
+    x_center = width * 0.5
+    y_center = height * 0.5
+    x_left = 0
+    x_right = width
+    y_top = 0
+    y_bottom = height
+
+    d_center_x = torch.pow(torch.arange(width) - x_center + 0.5, 2)
+    d_center_y = torch.pow(torch.arange(height) - y_center + 0.5, 2)
+    d_center = torch.sqrt(d_center_y.unsqueeze(0).transpose(0, 1) + d_center_x)
+
+    d_edge_left = torch.pow(torch.arange(width) - x_left + 0.5, 2)
+    d_edge_right = torch.pow(torch.arange(width) - x_right + 0.5, 2)
+    d_edge_top = torch.pow(torch.arange(height) - y_top + 0.5, 2)
+    d_edge_bottom = torch.pow(torch.arange(height) - y_bottom + 0.5, 2)
+
+    d_edge_x = torch.sqrt(torch.min(d_edge_left, d_edge_right) + 0.5 ** 2)
```

Comment:
`0.5 ** 2` == `0.25` isn't it? :)

Reply:
It's for readability. Maybe it'll be better to use torch.norm(..., p=2)